projects/vehicles/plugins/robot_windows/automobile/automobile.js

```diff
@@ -0,0 +1,237 @@
+  if (document.getElementById(name + '-section'))
+    return; // check if already exists
+
+  let div = '<div id="' + device.name + '-section" class="vehicle-device device">';
+  div += '<h2>';
+  div += '<input type="checkbox" title="Enable/disable this plot." id="' + device.name.toLowerCase() + '-enable-checkbox" device="' + device.name + '" onclick="vehicleCheckboxCallback(this)" />';
+  div += device.name + '<span id="' + device.name + '-label"></span></h2>';
+  div += '<div id="' + device.name + '-content" class="vehicle-device-content device-content"/>';
+  div += '</div>';
+
+  appendNewElement(device.name + '-layout', div);
+
+  let decimals = 3;
+  let plotLabels = {'x': 'Time [s]', 'y': label};
+  let autoRange = TimeplotWidget.prototype.AutoRangeType.STRETCH;
+  if (device.name.toLowerCase() === 'steering') {
+    plotLabels['legend'] = ['steering angle', 'right steering angle', 'left steering angle'];
+    decimals = 4;
+  } else if (device.name.toLowerCase() === 'encoders') {
+    plotLabels['legend'] = ['front right', 'front left', 'rear right', 'rear left'];
+    decimals = 1;
+  }
+
+  const widget = new VehicleTimeplotWidget(document.getElementById(device.name + '-content'), basicTimeStep, autoRange, {'min': min, 'max': max}, plotLabels, device, decimals);
+  widget.setLabel(document.getElementById(device.name + '-label'));
+  automobileWidgets[device.name.toLowerCase()] = widget;
+}
+
+function setDeviceModeCallback(switchButton, deviceType) {
+  const messageHeader = 'device-control-mode:' + deviceType;
+  const message = messageHeader + ':' + (switchButton.checked ? '1' : '0');
+  for (let i = 0; i < commands.length; ++i) {
+    if (commands[i].startsWith(messageHeader)) {
+      commands[i] = message;
+      return;
+    }
+  }
+  commands.push(message);
+
+  // force widgets refresh when they are shown.
+  Object.keys(window.widgets).forEach(function(deviceName) {
+    const widget = window.widgets[deviceName];
+    if (widget && widget.device.type === deviceType) {
+      const checkbox = document.getElementById(widget.device.name + '-enable-checkbox');
+      DeviceWidget.checkboxCallback(checkbox);
+    }
+  });
+}
+
+function configure(data) {
+  robotName = data.name;
+  basicTimeStep = data.basicTimeStep;
+
+  window.robotWindow.setTitle('Vehicle robot window [' + robotName + ']');
+
+  if (data.devices == null) {
+    document.getElementById('no-controller-label').innerHTML = 'No devices.';
+    closeMenu();
+    return;
+  }
+
+  document.getElementById('no-controller-label').style.display = 'none';
+
+  // Overview tab
+  overviewWidget = new OverviewWidget(document.getElementById('overview-section'));
+
+  addTab('Speed');
+  appendNewElement('Speed-layout', '<h2><span id="target-speed-label" style="color:red"></span></h2>');
+  addDriverInfo({name: 'Speed', htmlName: 'Speed'}, '[km/h]', -20.0, 20.0);
+
+  addTab('Steering');
+  addDriverInfo({name: 'Steering'}, '[rad]', -0.1, 0.1);
```

Comment:
This is not sufficient for the boomer and probably for most vehicles.
```suggestion
  addDriverInfo({name: 'Steering'}, '[rad]', -1, 1);
```

Reply:
In the `city.wbt` example the steering ranges from [-0.15, 0.1]. So a range [-1, 1] is extremely big in most of the real cases.

Personally I really prefer (and it is what all the graphs in the robot window are doing) to set a smaller initial range and let the graph resize automatically. Otherwise for most of the vehicles there will be a big range where only 10% of it is used and a very poor resolution. So we will produce bad and hardly readable graphs.

Reply:
I see... Can't we initialize these min and max value with the actual min and max position of the corresponding motors as we do in the default motor control widget?

Reply:
The problem with the HTML window and resizing is that it erases everything before resizing (unlike the Qt robot window) which makes it hard to view and understand what's going on...

Reply:
The min and max motor positions of the BMW X5 wheels are [-1, 1].
So if we should use these value, this range will be too big for a normal run of the city simulation.

Reply:
To me this is less annoying than having the graph resetting (and clearing all data) every time you increase or decrease the position outside of the current range.

Reply:
The problem is that if the graph is reset, data will be lost just for some frames. But if you set a bad range, then this will remains for the whole simulation.

I think that the real solution is to fix the graph update (in the generic robot winow) when the range changes and not hard coding the range based on the worst cases.

Just to information, this is the resulting graph for `city.wbt` when setting the range to [-1, 1] and it is very poor in comparison to what we can achieve automatically resizing the graph.
![city_steering](https://user-images.githubusercontent.com/5910449/109932262-42bf3f80-7cca-11eb-9d74-28a3cdfc6249.png)

Reply:
I agree that the best solution is to fix the graph update.
For most users, the broken graph update looks really buggy whereas the small amplitude of the graph within the min/max range is more understandable even if it is not ideal.